bpf/tests/Makefile

```diff
@@ -0,0 +1,48 @@
+# Copyright Authors of Cilium
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+
+include ../../Makefile.defs
+
+MAKEFLAGS += -r
+
+CLANG ?= clang
+LLC   ?= llc
+
+FLAGS := -I$(ROOT_DIR)/bpf/include -I$(ROOT_DIR)/bpf -D__NR_CPUS__=$(shell nproc --all) -O2 -g
+
+CLANG_FLAGS := ${FLAGS} -target bpf -std=gnu99 -nostdinc -emit-llvm
+# eBPF verifier enforces unaligned access checks where necessary, so don't
+# let clang complain too early.
+CLANG_FLAGS += -Wall -Wextra -Werror -Wshadow
+CLANG_FLAGS += -Wno-address-of-packed-member
+CLANG_FLAGS += -Wno-unknown-warning-option
+CLANG_FLAGS += -Wno-gnu-variable-sized-type-not-at-end
+CLANG_FLAGS += -Wdeclaration-after-statement
+CLANG_FLAGS += -Wimplicit-int-conversion -Wenum-conversion
+LLC_FLAGS := -march=bpf -mattr=dwarfris
```

Comment:
Do we need to be figuring out the clang flags independently for the test progs? Or can we reuse existing flags from the other makefiles? (even if we have to shift them to either `Makefile.defs` or `bpf/Makefile.bpf` or something).